airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/stream.py

```diff
@@ -410,11 +434,14 @@ def stream_slices(
             prev_file_last_mod: datetime = None  # init variable to hold previous iterations last modified
             grouped_files_by_time: List[Dict[str, Any]] = []
             for file_info in self.get_time_ordered_file_infos():
-                # skip this file if last_mod is earlier than our cursor value from state
+                # skip this file if last_mod is earlier than our cursor value from state and already in history
+                # or skip this file if last_mod plus delta is earlier than our cursor value
                 if (
                     stream_state is not None
                     and self.cursor_field in stream_state.keys()
                     and file_info.last_modified <= self._get_datetime_from_stream_state(stream_state)
+                    and self.file_in_history(file_info, stream_state.get("history", {}))
+                    or file_info.last_modified + timedelta(days=self.days) < self._get_datetime_from_stream_state(stream_state)
```

Comment:
could you add some parens to make it clear the order of operations on the ORs and ANDs? is it `(X and Y and Z) or A` or `X and Y and (Z or A)`?

Reply:
updated